driver/prestoresults.d

```
@@ -26,10 +26,17 @@ import presto.client.queryresults : ColumnMetadata;
 import presto.odbcdriver.bindings : OdbcResult, OdbcResultRow;
 import presto.odbcdriver.util : dllEnforce, logMessage;
 
-void addToPrestoResultRow(JSONValue columnData, PrestoResultRow result) {
+void addToPrestoResultRow(JSONValue columnData, PrestoResultRow result, string resultDataType) {
     final switch (columnData.type) {
         case JSON_TYPE.STRING:
-            result.addNextValue(Variant(columnData.str));
+            if (resultDataType == "string") {
+                result.addNextValue(Variant(columnData.str));
+            } else {
+                //for some scalar functions that return double, we may get a "NaN" back
+                if (columnData.str == "NaN") { 
+                    result.addNextValue(Variant(double.nan));
+                }
```

Should we throw an exception in the `else` case?